src/include/interval_set.h

```diff
@@ -249,44 +249,60 @@ class interval_set {
  private:
   auto find_inc(T start) const {
     auto p = m.lower_bound(start);  // p->first >= start
-    if (p != m.begin() &&
-        (p == m.end() || p->first > start)) {
-      --p;   // might overlap?
-      if (p->first + p->second <= start)
-        ++p; // it doesn't.
+    while (p != m.begin()) {
+        if (p->first <= start) {
```

Comment:
This is buggy. Old code will make sure that the `p` won't be `.end()`, but yours seems will crash ?

Reply:
so if we dont hit this `if {}` then it assume either `p->first > start` or `p==m.end()`, so in this case we decrement `--p`

Reply:
i.e. it wont hit `end()` here either

Reply:
Won't the `m.lower_bound()` return `.end()` if it couldn't find any item ?

Reply:
Yeah, the corner edge case.

Reply:
check now

Reply:
In `erase()` it will check `ceph_assert(p != m.end()); `. Your above change could return `.end()`.